### PR TITLE
Xml schema dependency removal for AOT compilers

### PR DESCRIPTION
**Pull request by vitek-karas**
This changes fixes a size issue in AOT compiled code which uses XmlReader. Using XmlReader for just reading XML without validation still pulls in all of the XmlSchema code. This makes the AOT compilation slow (it's more than 400KB of IL) and the resulting app is larger (.NET Native X86 measures the difference at more than 800KB of native code).
The change relies on an assumption, that apps which don't need validation will not call the XmlReaderSettings.ValidationType setter. As such it can be reduced away and the change then ties all the dependencies needed for validation to that setter. So if the setter itself is not called XmlSchema is not pulled in.
Verified that the change does reduce the app size and that XmlSchema is not in final app anymore.

### CODE REVIEW

File: src/System.Private.Xml/src/System/Xml/Serialization/XmlSerializer.cs [hunk 1]
```diff
@@ -115,25 +115,29 @@ public abstract class XmlSerializerImplementation
         public virtual XmlSerializer GetSerializer(Type type) { throw new NotSupportedException(); }
     }
 
+    // This enum is intentionally kept outside of the XmlSerializer class since if it would be a subclass
+    // of XmlSerializer, then any access to this enum would be treated by AOT compilers as access to the XmlSerializer
+    // as well, which has a large static ctor which brings in a lot of code. So keeping the enum separate
+    // makes sure that using just the enum itself doesn't bring in the whole of serialization code base.
+#if FEATURE_SERIALIZATION_UAPAOT
+    public enum SerializationMode
```
Comment (krwq) [inline src/System.Private.Xml/src/System/Xml/Serialization/XmlSerializer.cs]: are there any compat concerns regarding moving this type outside?

Comment (krwq) [reply src/System.Private.Xml/src/System/Xml/Serialization/XmlSerializer.cs]: ahh this is only public for UAPAOT

Comment (vitek-karas) [reply src/System.Private.Xml/src/System/Xml/Serialization/XmlSerializer.cs]: As you already noted that is basically a private test hook. We had to make it public for UAPAOT to work correctly, but it should only be used by tests.

**Comment by krwq** [comment]
test CROSS Check

**Comment by krwq** [comment]
@dotnet-bot test this please

**Comment by krwq** [comment]
@dotnet-bot help

**Pull request merged**